src/entities/Basket/Basket.controller.ts

```diff
@@ -0,0 +1,20 @@
+import StatusCodes from 'http-status-codes';
+import { Request, Response } from 'express';
+import validateBasketReq from './Basket.validation';
+import Basket from './Basket.schema'
+
+const { BAD_REQUEST, CREATED, OK } = StatusCodes;
+
+export const addBasket = async (req: Request, res: Response) => {
+    const { error } = validateBasketReq(req.body);
+    if (error) return res.status(BAD_REQUEST).send(error.details[0].message);
+    const basketData = req.body;
+    const createdByUserId = {createdByUserId: req.user}
+    const basket = new Basket({...basketData, ...createdByUserId})
+    try {
+        await basket.save()
+        return res.status(CREATED).json(basket);
+    } catch (error) {
+        return res.status(BAD_REQUEST).send(error._message);
+    }
+}
```

Comment:
It looks good to me, but we should add some validation of the books that are being put in the basket - we need to check if the books belong to the users who offer/request them :)

Reply:
Validation was added in -> 7e1628749af4bebe48c4722f8fb5f5b27c68c572,

Reply:
I haven't checked if it works yet (it probably does :D ) but I think you could move this code to the separate function in the same file (to the top)

Reply:
Btw the rest looks fine to me ;)

Reply:
DONE -> 72ffa7996652f1f5bcee0b7e1a21a1b105e98d50